test/autonat/index.spec.ts

```diff
@@ -92,19 +96,28 @@ describe('autonat', () => {
       connection.remoteAddr = multiaddr(`/ip4/${host}/tcp/28319/p2p/${peer.id.toString()}`)
       connectionManager.openConnection.withArgs(peer.id).resolves(connection)
 
-      // stub autonat protocol stream
-      const stream = stubInterface<Stream>()
-      connection.newStream.withArgs(`/${PROTOCOL_PREFIX}/${PROTOCOL_NAME}/${PROTOCOL_VERSION}`).resolves(stream)
-
-      // stub autonat response
       const response = Message.encode({
         type: Message.MessageType.DIAL_RESPONSE,
         dialResponse
       })
-      stream.source = (async function * () {
-        yield lp.encode.single(response)
-      }())
-      stream.sink.returns(Promise.resolve())
+
+      // stub autonat protocol stream
+      const stream = stubObject<Stream>({
```

Comment:
Trying to run this branch locally I noticed something weird with the stubbing but only in WebKit, though I didn't get to the bottom of it.

We stub a bunch of peer responses in each test to simulate different scenarios - once a certain number of tests were being run, the stubbed return values started being from the wrong tests.

Needs more investigation really.

Reply:
Yes. Hence this change. I'm not sure why exactly this fails on WebKit but seems to work on other browsers. I'm not sure overriding values is supported in Sinon or if it's a hack we do. The change here is to not override and instead be explicit in what gets stubbed vs set.

My guess is the other failures we're seeing is related to something like this.

Reply:
Have you checked out the Sinon docs? I believe we’re using it as intended, no “hack” involved.

Reply:
I think sinon is not involved here. This should just be setting values on the object here: https://github.com/ttarnowski/ts-sinon/blob/master/src/index.ts#L67. Maybe `Proxy` in WebKit has slightly different behavior? Maybe a bug since this only shows up in certain cases (e.g. run only this test is fine, but run two versions of this test fails).

Reply:
Yeah, it’s really weird because we don’t reuse the stubbed instances between the tests that I can see 🤔

Reply:
@maschad did you figure out what the issue was here around Proxy?

Reply:
@MarcoPolo I don't think the issue was related to `Proxy` but rather the peer id fixtures, the relay server behaved as if it was creating reservations for peers that _seemed_ to be reconnecting due to the tests reusing peer ids, we resolved this [in this PR](https://github.com/libp2p/js-libp2p/pull/1795)